### DISCUSSION TITLE
Mason language filter not working as it is supposed to?

### DISCUSSION THREAD

**Discussion by FosRexx** [NvChad/NvChad#2939] [Q&A] [answered]
So, I was just trying to install some formatter and so when i tried to search for the language using the <C-f> keybind in mason, the filter was not interactive. The filter looks like this:
![image](https://github.com/user-attachments/assets/1c5c43d3-bee5-4549-ac30-d8b759c42cc1)

Then i looked into mason's github and saw this screenshot:
![image](https://github.com/user-attachments/assets/4ef14d7d-4e9c-4aa2-9bf1-3401610af8e8)

The one in NVChad is not interactive, I dont understand why, I have not explicitly changed any configuration for mason, that I know of.
Is this intentional? Can I make the filter interactive?

**Comment by siduck** [comment]
the one in your pic isnt even from mason, its some plugin which converts the list for telescope

**Reply by FosRexx** [reply]
Here are the installed plugins:
![nvim_plugins](https://github.com/user-attachments/assets/85ea1f54-5751-400d-a099-a87da68b19d4)


Here is me trying to use language filter in mason:

https://github.com/user-attachments/assets/b5b8dff6-0688-4bc9-aa13-dc700e1c4c99

**Reply by siduck** [reply, marked as answer]
@FosRexx dressing.nvim is needed

```lua
{
  'stevearc/dressing.nvim',
  opts = {},
 event='VeryLazy'
}
```

**Reply by theunhackable** [reply]
this should be in the docs. Thank you @siduck

**Reply by siduck** [reply]
@theunhackable i dont think so, you can use the mason filter without dressing plugin too

**Reply by Sourabhpatel1** [reply]
How nothing in docs and nothing in ui?

**Reply by siduck** [reply]
@Sourabhpatel1 wym  nothing in ui? press ctrl-f and you'll see a huge list, printed. just type the number

**Comment by Snake-Host** [comment]
dressing.nvim repo recommends switching to snacks.nvim, I've installed snacks but the same problem remains. Any idea on how to achieve the same results using snacks instead of dressing?

**Reply by siduck** [reply]
@Snake-Host  dont layzload snack, do lazy=false

**Reply by Snake-Host** [reply]
I installed snacks using neovim native package manager using `vim.pack.add{ "https://github.com/folke/snacks.nvim" }`

**Reply by siduck** [reply]
> I installed snacks using neovim native package manager using `vim.pack.add{ "https://github.com/folke/snacks.nvim" }`

nvchad doesnt support package manager other than lazy.nvim
